dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/converter/EnrollmentTrackerConverterService.java

```diff
@@ -148,8 +149,8 @@ public ProgramInstance from( TrackerPreheat preheat, Enrollment enrollment )
                 programInstance.setUid( CodeGenerator.generateUid() );
             }
 
-            programInstance.setEnrollmentDate( enrollment.getEnrollmentDate() );
-            programInstance.setIncidentDate( enrollment.getIncidentDate() );
+            programInstance.setEnrollmentDate( DateUtils.parseDate(enrollment.getEnrollmentDate()) );
```

Comment:
I'm doing something similar in my DTO branch.. we might have a merge conflict here (but the approach is very similar)